pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test: TestSQLWatcherOnEventError failed

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4407959&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4407959&tab=artifacts#/) on master @ [1113ce6c6b53147821fd808a0ca215ef81509ee4](https://github.com/cockroachdb/cockroach/commits/1113ce6c6b53147821fd808a0ca215ef81509ee4):


```
=== RUN   TestSQLWatcherOnEventError
    sqlwatcher_test.go:437: handler should never run
--- FAIL: TestSQLWatcherOnEventError (0.96s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-13266

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4566076&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4566076&tab=artifacts#/) on master @ [231fd420c61e39b6ad08c982c496d82cf1910bc5](https://github.com/cockroachdb/cockroach/commits/231fd420c61e39b6ad08c982c496d82cf1910bc5):


```
=== RUN   TestSQLWatcherOnEventError
    sqlwatcher_test.go:437: handler should never run
--- FAIL: TestSQLWatcherOnEventError (0.98s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4962781&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4962781&tab=artifacts#/) on master @ [6537583259df42cb98f6301e969089e5064df614](https://github.com/cockroachdb/cockroach/commits/6537583259df42cb98f6301e969089e5064df614):


```
=== RUN   TestSQLWatcherOnEventError
    sqlwatcher_test.go:454: handler should never run
--- FAIL: TestSQLWatcherOnEventError (0.80s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=5061315&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=5061315&tab=artifacts#/) on master @ [a2e1910f51593bd2ef72e1d7c615e08f95791186](https://github.com/cockroachdb/cockroach/commits/a2e1910f51593bd2ef72e1d7c615e08f95791186):


```
=== RUN   TestSQLWatcherOnEventError
    sqlwatcher_test.go:454: handler should never run
--- FAIL: TestSQLWatcherOnEventError (0.94s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=5149202&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=5149202&tab=artifacts#/) on master @ [bc1ee7c7c276984fce8ff5ba4fcfcdff335dde50](https://github.com/cockroachdb/cockroach/commits/bc1ee7c7c276984fce8ff5ba4fcfcdff335dde50):


```
=== RUN   TestSQLWatcherOnEventError
    sqlwatcher_test.go:454: handler should never run
--- FAIL: TestSQLWatcherOnEventError (1.09s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=5164266&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=5164266&tab=artifacts#/) on master @ [bc5f5b7159ed2b7dbffa94bb2a590fdddb642b9b](https://github.com/cockroachdb/cockroach/commits/bc5f5b7159ed2b7dbffa94bb2a590fdddb642b9b):


```
=== RUN   TestSQLWatcherOnEventError
    sqlwatcher_test.go:454: handler should never run
--- FAIL: TestSQLWatcherOnEventError (0.85s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=5190497&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=5190497&tab=artifacts#/) on master @ [a35b848edb3f5562199b0f56bc27a6c576073d5d](https://github.com/cockroachdb/cockroach/commits/a35b848edb3f5562199b0f56bc27a6c576073d5d):


```
=== RUN   TestSQLWatcherOnEventError
    sqlwatcher_test.go:454: handler should never run
--- FAIL: TestSQLWatcherOnEventError (0.86s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=5202719&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=5202719&tab=artifacts#/) on master @ [ce66acdbba801de88f1dd645eaedeb3834f23dbd](https://github.com/cockroachdb/cockroach/commits/ce66acdbba801de88f1dd645eaedeb3834f23dbd):


```
=== RUN   TestSQLWatcherOnEventError
    sqlwatcher_test.go:454: handler should never run
--- FAIL: TestSQLWatcherOnEventError (0.96s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/buildConfiguration/BUILDTYPE_ID-not-found-in-env/5213887?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/BUILDTYPE_ID-not-found-in-env/5213887?buildTab=artifacts#/) on master @ [92947c29c55ff909f50a5e625811d34a1bbe71f7](https://github.com/cockroachdb/cockroach/commits/92947c29c55ff909f50a5e625811d34a1bbe71f7):


```
=== RUN   TestSQLWatcherOnEventError
    sqlwatcher_test.go:454: handler should never run
--- FAIL: TestSQLWatcherOnEventError (0.84s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5239277?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5239277?buildTab=artifacts#/) on master @ [5c400e31f33b97a7b3232bb54abdd55645f0e978](https://github.com/cockroachdb/cockroach/commits/5c400e31f33b97a7b3232bb54abdd55645f0e978):


```
=== RUN   TestSQLWatcherOnEventError
    sqlwatcher_test.go:454: handler should never run
--- FAIL: TestSQLWatcherOnEventError (1.03s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5336597?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5336597?buildTab=artifacts#/) on master @ [1cea73c8a18623949b81705eb5f75179e6cd8d86](https://github.com/cockroachdb/cockroach/commits/1cea73c8a18623949b81705eb5f75179e6cd8d86):


```
=== RUN   TestSQLWatcherOnEventError
    sqlwatcher_test.go:454: handler should never run
--- FAIL: TestSQLWatcherOnEventError (0.94s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82031 pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test: TestSQLWatcherOnEventError failed [C-test-failure O-robot branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5348504?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5348504?buildTab=artifacts#/) on master @ [5a54758ce89e866b4fe28c0df74bd610973c6918](https://github.com/cockroachdb/cockroach/commits/5a54758ce89e866b4fe28c0df74bd610973c6918):


```
=== RUN   TestSQLWatcherOnEventError
    sqlwatcher_test.go:454: handler should never run
--- FAIL: TestSQLWatcherOnEventError (0.95s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82031 pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test: TestSQLWatcherOnEventError failed [C-test-failure O-robot branch-release-22.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5420170?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5420170?buildTab=artifacts#/) on master @ [3706c07f7332098d3eb0fcb6ae3b682a385b1306](https://github.com/cockroachdb/cockroach/commits/3706c07f7332098d3eb0fcb6ae3b682a385b1306):


```
=== RUN   TestSQLWatcherOnEventError
    sqlwatcher_test.go:454: handler should never run
--- FAIL: TestSQLWatcherOnEventError (1.12s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82031 pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test: TestSQLWatcherOnEventError failed [C-test-failure O-robot branch-release-22.1 sync-me sync-me-5]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5442768?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5442768?buildTab=artifacts#/) on master @ [45ee8026c1997df738ee184fe6a606e927c0a8c5](https://github.com/cockroachdb/cockroach/commits/45ee8026c1997df738ee184fe6a606e927c0a8c5):


```
=== RUN   TestSQLWatcherOnEventError
    sqlwatcher_test.go:454: handler should never run
--- FAIL: TestSQLWatcherOnEventError (0.99s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82031 pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test: TestSQLWatcherOnEventError failed [C-test-failure O-robot branch-release-22.1 sync-me sync-me-5]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5454272?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5454272?buildTab=artifacts#/) on master @ [d82ac30cc7b3cb5566e347ae1fecd2025c5e0623](https://github.com/cockroachdb/cockroach/commits/d82ac30cc7b3cb5566e347ae1fecd2025c5e0623):


```
=== RUN   TestSQLWatcherOnEventError
    sqlwatcher_test.go:454: handler should never run
--- FAIL: TestSQLWatcherOnEventError (1.06s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82031 pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test: TestSQLWatcherOnEventError failed [C-test-failure O-robot branch-release-22.1 sync-me sync-me-5]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5500783?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5500783?buildTab=artifacts#/) on master @ [8d34ef1ea15850ee1c70470610b6652df4c317de](https://github.com/cockroachdb/cockroach/commits/8d34ef1ea15850ee1c70470610b6652df4c317de):


```
=== RUN   TestSQLWatcherOnEventError
    sqlwatcher_test.go:460: handler should never run
--- FAIL: TestSQLWatcherOnEventError (1.19s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82031 pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test: TestSQLWatcherOnEventError failed [C-test-failure O-robot branch-release-22.1 sync-me sync-me-5]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5511242?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5511242?buildTab=artifacts#/) on master @ [8d34ef1ea15850ee1c70470610b6652df4c317de](https://github.com/cockroachdb/cockroach/commits/8d34ef1ea15850ee1c70470610b6652df4c317de):


```
=== RUN   TestSQLWatcherOnEventError
    sqlwatcher_test.go:460: handler should never run
--- FAIL: TestSQLWatcherOnEventError (1.12s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82031 pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test: TestSQLWatcherOnEventError failed [C-test-failure O-robot branch-release-22.1 sync-me sync-me-5]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5517551?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/5517551?buildTab=artifacts#/) on master @ [377b113bd7e7b679c55a0f93e0d8ab5dbb6f2e97](https://github.com/cockroachdb/cockroach/commits/377b113bd7e7b679c55a0f93e0d8ab5dbb6f2e97):


```
=== RUN   TestSQLWatcherOnEventError
    sqlwatcher_test.go:460: handler should never run
--- FAIL: TestSQLWatcherOnEventError (1.07s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #82031 pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test: TestSQLWatcherOnEventError failed [C-test-failure O-robot branch-release-22.1 sync-me sync-me-5]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8764541?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8764541?buildTab=artifacts#/) on master @ [e9c96e7179e19aae2f8d386f67eb950db8c3354b](https://github.com/cockroachdb/cockroach/commits/e9c96e7179e19aae2f8d386f67eb950db8c3354b):


```
W230220 07:28:32.799391 1307734 sql/sqlliveness/slinstance/slinstance.go:312  [T1,n1] 797  exiting heartbeat loop
I230220 07:28:32.797312 1315118 sql/stats/automatic_stats.go:507  [T1,n1] 794  quiescing auto stats refresher
I230220 07:28:32.799058 1298508 server/start_listen.go:103  [T1,n1] 798  server shutting down: instructing cmux to stop accepting
W230220 07:28:32.804331 1300705 kv/kvserver/replica_proposal.go:505  [T1,n1,s1,r34/1:/Table/3{2-3},raft] 799  lease <empty> expired before being followed by lease repl=(n1,s1):1 seq=1 start=0,0 epo=1 pro=1676878112.792404825,0; foreground traffic may have been impacted
W230220 07:28:32.805555 1307734 sql/sqlliveness/slinstance/slinstance.go:297  [T1,n1] 800  exiting heartbeat loop
E230220 07:28:32.806190 1307734 server/server_sql.go:474  [T1,n1] 801  failed to run update of instance with new session ID: node unavailable; try another peer
E230220 07:28:32.806571 1315294 jobs/registry.go:994  [T1,n1] 802  error expiring job sessions: expire-sessions: failed to read query result: context canceled
E230220 07:28:32.807569 1315294 jobs/registry.go:1002  [T1,n1] 803  failed to serve pause and cancel requests: txn exec: context canceled
E230220 07:28:32.807776 1315336 jobs/registry.go:1062  [T1,n1] 804  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: failed to read query result: context canceled
W230220 07:28:32.807820 1315294 jobs/registry.go:1078  [T1,n1] 805  canceling all adopted jobs due to stopper quiescing
E230220 07:28:32.808298 1542216 jobs/registry.go:950  [T1,n1] 806  error getting live session: node unavailable; try another peer
E230220 07:28:32.810376 1574295 kv/kvclient/kvcoord/txn_interceptor_committer.go:489  [T1] 807  making txn commit explicit failed for "sql txn" meta={id=bd21503c key=/Table/3/1/104/2/1 pri=0.00547436 epo=0 ts=1676878112.695089141,0 min=1676878112.695089141,0 seq=3} lock=true stat=STAGING rts=1676878112.695089141,0 wto=false gul=1676878113.195089141,0 ifw=2: result is ambiguous: server shutdown
W230220 07:28:32.810583 1574342 1@sql/event_log.go:653  [T1,n1,client=127.0.0.1:47642,user=root] 808  unable to save 1 entries to system.eventlog: log-event: failed to read query result: context canceled
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809  failed node liveness heartbeat: txn exec: context canceled
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +(1) attached stack trace
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  -- stack trace:
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | 	github.com/cockroachdb/cockroach/pkg/kv/txn.go:926
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | github.com/cockroachdb/cockroach/pkg/kv.runTxn
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:959
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | github.com/cockroachdb/cockroach/pkg/kv.(*DB).TxnWithAdmissionControl
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:922
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:897
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/liveness.(*NodeLiveness).updateLivenessAttempt
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/liveness/liveness.go:1343
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/liveness.(*NodeLiveness).updateLiveness
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/liveness/liveness.go:1298
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/liveness.(*NodeLiveness).heartbeatInternal
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/liveness/liveness.go:965
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/liveness.(*NodeLiveness).Start.func1.1
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/liveness/liveness.go:780
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | 	github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:91
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/liveness.(*NodeLiveness).Start.func1
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/liveness/liveness.go:763
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:470
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | runtime.goexit
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +  | 	GOROOT/src/runtime/asm_amd64.s:1594
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +Wraps: (2) txn exec
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +Wraps: (3) context canceled
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *errors.errorString
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +An inability to maintain liveness will prevent a node from participating in a
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +cluster. If this problem persists, it may be a sign of resource starvation or
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +of network connectivity problems. For help troubleshooting, visit:
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +
W230220 07:28:32.811776 1301137 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 809 +    https://www.cockroachlabs.com/docs/stable/cluster-setup-troubleshooting.html#node-liveness-issues
--- FAIL: TestSQLWatcherOnEventError (3.29s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #97319 pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test: TestSQLWatcherOnEventError failed [C-test-failure O-robot T-kv branch-release-22.2]
- #82031 pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test: TestSQLWatcherOnEventError failed [C-test-failure O-robot branch-release-22.1 sync-me sync-me-5]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9012319?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9012319?buildTab=artifacts#/) on master @ [5b2a5670cbbe895d76602c230390816e783e0caa](https://github.com/cockroachdb/cockroach/commits/5b2a5670cbbe895d76602c230390816e783e0caa):


```
I230311 07:14:02.617652 1350351 jobs/wait.go:142  [T1,n1] 671  waited for 1 [846883855870001153] queued jobs to complete 247.197513ms
I230311 07:14:02.619308 1350351 upgrade/upgrademanager/manager.go:268  [T1,n1] 672  running permanent upgrade for version 1000022.2-38
I230311 07:14:02.650941 1583069 jobs/registry.go:1495  [T1,n1] 673  KEY VISUALIZER job 100: stepping through state running
I230311 07:14:02.737624 1350351 upgrade/upgrademanager/manager.go:584  [T1,n1] 674  running Upgrade to 1000022.2-38: "create jobs metrics polling job"
I230311 07:14:02.737826 1593481 jobs/adopt.go:247  [T1,n1] 675  job 846883856775249921: resuming execution
I230311 07:14:02.773918 1595581 jobs/registry.go:1495  [T1,n1] 676  MIGRATION job 846883856775249921: stepping through state running
I230311 07:14:02.852662 1595581 jobs/registry.go:1495  [T1,n1] 677  MIGRATION job 846883856775249921: stepping through state succeeded
I230311 07:14:02.867549 1375149 jobs/adopt.go:110  [T1,n1] 678  claimed 1 jobs
I230311 07:14:02.894566 1350351 jobs/wait.go:142  [T1,n1] 679  waited for 1 [846883856775249921] queued jobs to complete 156.755366ms
I230311 07:14:02.894686 1350351 server/server_sql.go:1540  [T1,n1] 680  done ensuring all necessary startup migrations have run
I230311 07:14:02.899377 1608380 jobs/job_scheduler.go:407  [T1,n1] 681  waiting 4m0s before scheduled jobs daemon start
I230311 07:14:02.906811 1609546 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 682  tenant-settings-watcher: established range feed cache
I230311 07:14:02.914770 1610702 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 683  tenant-capability-watcher: established range feed cache
I230311 07:14:02.915057 1610799 jobs/adopt.go:247  [T1,n1] 684  job 101: resuming execution
I230311 07:14:02.927235 1610212 13@kv/kvserver/replicate_queue.go:819  [T1,n1,replicate,s1,r58/1:/{Table/57-Max}] 685  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230311 07:14:02.930231 1612855 13@kv/kvserver/replicate_queue.go:819  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 686  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230311 07:14:02.933221 1610702 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:147  [T1,n1] 687  received results of a full table scan for tenant capabilities
I230311 07:14:02.939414 1608497 server/auto_upgrade.go:78  [T1,n1] 688  no need to upgrade, cluster already at the newest version
I230311 07:14:02.963371 1617083 13@kv/kvserver/replicate_queue.go:819  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 689  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230311 07:14:02.971188 1608486 sql/syntheticprivilegecache/cache.go:195  [T1,n1] 690  warmed privileges for virtual tables in 70.039203ms
I230311 07:14:02.992690 1618252 jobs/registry.go:1495  [T1,n1] 691  POLL JOBS STATS job 101: stepping through state running
I230311 07:14:03.039319 1625383 13@kv/kvserver/replicate_queue.go:819  [T1,n1,replicate,s1,r4/1:/System{/tsd-tse}] 692  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230311 07:14:03.064738 1358026 3@pebble/event.go:694  [T1,n1,s1,pebble] 693  [JOB 6] WAL created 000005
I230311 07:14:03.076032 1350351 1@server/server_sql.go:1627  [T1,n1] 694  serving sql connections
I230311 07:14:03.077388 1350351 testutils/testcluster/testcluster.go:1338  [T1] 695  WaitForFullReplication
I230311 07:14:03.077444 1350351 testutils/testcluster/testcluster.go:1342  [T1] 696  WaitForFullReplication took: 296ns
I230311 07:14:03.116459 1628549 util/log/event_log.go:32  [T1,n1,client=127.0.0.1:37824,user=root] 697 ={"Timestamp":1678518843100772625,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"kv.rangefeed.enabled\" = true","Tag":"SET CLUSTER SETTING","User":"root","SettingName":"kv.rangefeed.enabled","Value":"true"}
I230311 07:14:03.152803 1628549 util/log/event_log.go:32  [T1,n1,client=127.0.0.1:37824,user=root] 698 ={"Timestamp":1678518843138527355,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"kv.closed_timestamp.target_duration\" = '100ms'","Tag":"SET CLUSTER SETTING","User":"root","SettingName":"kv.closed_timestamp.target_duration","Value":"00:00:00.1"}
I230311 07:14:03.244013 1638877 13@kv/kvserver/replicate_queue.go:819  [T1,n1,replicate,s1,r7/1:/Table/{3-4}] 699  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230311 07:14:03.275555 1628549 5@util/log/event_log.go:32  [T1,n1,client=127.0.0.1:37824,user=root] 700 ={"Timestamp":1678518843155087733,"EventType":"create_table","Statement":"CREATE TABLE defaultdb.public.t ()","Tag":"CREATE TABLE","User":"root","DescriptorID":104,"TableName":"defaultdb.public.t"}
I230311 07:14:03.276951 1350351 spanconfig/spanconfigsqlwatcher/sqlwatcher.go:245  [T1] 701  established range feed over system.descriptors starting at time 1678518843.154843113,0
I230311 07:14:03.277138 1350351 spanconfig/spanconfigsqlwatcher/sqlwatcher.go:304  [T1] 702  established range feed over system.zones starting at time 1678518843.154843113,0
I230311 07:14:03.281844 1350351 spanconfig/spanconfigsqlwatcher/sqlwatcher.go:400  [T1] 703  established range feed over system.protected_ts_records starting at time 1678518843.154843113,0
    sqlwatcher_test.go:460: handler should never run
W230311 07:14:03.282505 1641306 spanconfig/spanconfigsqlwatcher/sqlwatcher.go:133  [T1,rangefeed=sql-watcher-descriptor-rangefeed] 704  error adding event {{1678518843155087733 0 false} {{104 relation} {false {0}}}}: boom
I230311 07:14:03.287030 1350351 testutils/testcluster/testcluster.go:148  [T1] 705  TestCluster quiescing nodes
I230311 07:14:03.288532 1583069 jobs/registry.go:1495  [T1,n1] 706  KEY VISUALIZER job 100: stepping through state succeeded
I230311 07:14:03.290247 1355838 server/start_listen.go:103  [T1,n1] 707  server shutting down: instructing cmux to stop accepting
W230311 07:14:03.290590 1370436 sql/sqlliveness/slinstance/slinstance.go:312  [T1,n1] 708  exiting heartbeat loop
I230311 07:14:03.287731 1374857 sql/stats/automatic_stats.go:507  [T1,n1] 709  quiescing auto stats refresher
I230311 07:14:03.291615 1455668 rpc/context.go:2249  [T1,n1,rnode=1,raddr=127.0.0.1:36951,class=default,rpc] 710  connection heartbeat loop ended with err: <nil>
W230311 07:14:03.291101 1370436 sql/sqlliveness/slinstance/slinstance.go:297  [T1,n1] 711  exiting heartbeat loop
E230311 07:14:03.297490 1370436 server/server_sql.go:477  [T1,n1] 712  failed to run update of instance with new session ID: node unavailable; try another peer
W230311 07:14:03.289016 1635212 kv/kvserver/intentresolver/intent_resolver.go:826  [T1] 713  failed to gc transaction record: could not GC completed transaction anchored at /Table/6/1/"kv.closed_timestamp.target_duration"/0: node unavailable; try another peer
W230311 07:14:03.300142 1641235 1@sql/event_log.go:653  [T1,n1,client=127.0.0.1:37824,user=root] 714  unable to save 1 entries to system.eventlog: log-event: failed to read query result: query execution canceled
E230311 07:14:03.303726 1375132 jobs/registry.go:1004  [T1,n1] 715  failed to serve pause and cancel requests: could not query jobs table: cancel/pause-requested: failed to read query result: query execution canceled
W230311 07:14:03.304153 1375132 jobs/registry.go:1080  [T1,n1] 716  canceling all adopted jobs due to stopper quiescing
E230311 07:14:03.304506 1618252 jobs/registry.go:952  [T1,n1] 717  error getting live session: node unavailable; try another peer
E230311 07:14:03.305435 1375149 jobs/registry.go:1064  [T1,n1] 718  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: failed to read query result: query execution canceled
--- FAIL: TestSQLWatcherOnEventError (3.41s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #97319 pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test: TestSQLWatcherOnEventError failed [C-test-failure O-robot T-kv branch-release-22.2]
- #82031 pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test: TestSQLWatcherOnEventError failed [C-test-failure O-robot branch-release-22.1 sync-me sync-me-5]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9020233?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9020233?buildTab=artifacts#/) on master @ [95684388bfbb372ac8c7e1a8ebd4a0d6f447e147](https://github.com/cockroachdb/cockroach/commits/95684388bfbb372ac8c7e1a8ebd4a0d6f447e147):


```
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:322
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:233
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | github.com/cockroachdb/cockroach/pkg/rpc.bindUnaryServerInterceptorToHandler.func1
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:938
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func2
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:837
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor.ClientInterceptor.func2
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | 	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:227
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | github.com/cockroachdb/cockroach/pkg/rpc.getChainUnaryInvoker.func1
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:1022
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | github.com/cockroachdb/cockroach/pkg/rpc.makeInternalClientAdapter.func3
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:907
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | github.com/cockroachdb/cockroach/pkg/rpc.internalClientAdapter.Batch
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:1030
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).sendBatch
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/transport.go:210
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*grpcTransport).SendNext
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/transport.go:189
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendToReplicas
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender.go:2142
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendPartialBatch
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender.go:1668
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).divideAndSendBatchToRanges
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender.go:1239
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).Send
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender.go:860
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnLockGatekeeper).SendLocked
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_lock_gatekeeper.go:82
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnMetricRecorder).SendLocked
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_metric_recorder.go:47
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnCommitter).SendLocked
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_committer.go:203
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +Wraps: (2) aborted before proposing
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +Wraps: (3) context canceled
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +Error types: (1) *errbase.opaqueWrapper (2) *errutil.withPrefix (3) *errors.errorString
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +An inability to maintain liveness will prevent a node from participating in a
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +cluster. If this problem persists, it may be a sign of resource starvation or
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +of network connectivity problems. For help troubleshooting, visit:
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +
W230312 07:21:13.485826 1355808 kv/kvserver/liveness/liveness.go:792  [T1,n1,liveness-hb] 727 +    https://www.cockroachlabs.com/docs/stable/cluster-setup-troubleshooting.html#node-liveness-issues
E230312 07:21:13.486786 1371047 jobs/registry.go:1064  [T1,n1] 728  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: failed to read query result: query execution canceled
E230312 07:21:13.489897 1370981 jobs/registry.go:996  [T1,n1] 729  error expiring job sessions: expire-sessions: failed to read query result: query execution canceled
E230312 07:21:13.490147 1370981 jobs/registry.go:1004  [T1,n1] 730  failed to serve pause and cancel requests: txn exec: context canceled
W230312 07:21:13.490273 1370981 jobs/registry.go:1080  [T1,n1] 731  canceling all adopted jobs due to stopper quiescing
E230312 07:21:13.490553 1607679 jobs/registry.go:952  [T1,n1] 732  error getting live session: node unavailable; try another peer
W230312 07:21:13.507231 1633896 1@sql/event_log.go:653  [T1,n1,client=127.0.0.1:56012,user=root] 733  unable to save 1 entries to system.eventlog: log-event: failed to read query result: query execution canceled
--- FAIL: TestSQLWatcherOnEventError (3.34s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #97319 pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test: TestSQLWatcherOnEventError failed [C-test-failure O-robot T-kv branch-release-22.2]
- #82031 pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test: TestSQLWatcherOnEventError failed [C-test-failure O-robot branch-release-22.1 sync-me sync-me-5]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9051227?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9051227?buildTab=artifacts#/) on master @ [43f7cde6f036f720388912bde7310095c2f8ca9e](https://github.com/cockroachdb/cockroach/commits/43f7cde6f036f720388912bde7310095c2f8ca9e):


```
I230314 06:17:56.053794 2148143 jobs/registry.go:1489  [T1,n1] 665  KEY VISUALIZER job 100: stepping through state running
I230314 06:17:56.123654 1823186 jobs/wait.go:142  [T1,n1] 666  waited for 1 [847722169889030145] queued jobs to complete 562.992225ms
I230314 06:17:56.125298 1823186 upgrade/upgrademanager/manager.go:268  [T1,n1] 667  running permanent upgrade for version 1000022.2-38
I230314 06:17:56.233229 1823186 upgrade/upgrademanager/manager.go:584  [T1,n1] 668  running Upgrade to 1000022.2-38: "create jobs metrics polling job"
I230314 06:17:56.233405 2169610 jobs/adopt.go:247  [T1,n1] 669  job 847722172008005633: resuming execution
I230314 06:17:56.245503 2170948 jobs/adopt.go:247  [T1,n1] 670  job 847722172008005633: resuming execution
I230314 06:17:56.314671 2171723 jobs/registry.go:1489  [T1,n1] 671  MIGRATION job 847722172008005633: stepping through state running
I230314 06:17:56.435489 1845359 jobs/adopt.go:110  [T1,n1] 672  claimed 1 jobs
I230314 06:17:56.446936 2171723 jobs/registry.go:1489  [T1,n1] 673  MIGRATION job 847722172008005633: stepping through state succeeded
I230314 06:17:56.481155 2193120 jobs/adopt.go:247  [T1,n1] 674  job 101: resuming execution
I230314 06:17:56.576648 1823186 jobs/wait.go:142  [T1,n1] 675  waited for 1 [847722172008005633] queued jobs to complete 343.259367ms
I230314 06:17:56.576776 1823186 server/server_sql.go:1540  [T1,n1] 676  done ensuring all necessary startup migrations have run
I230314 06:17:56.579181 2202498 jobs/job_scheduler.go:407  [T1,n1] 677  waiting 2m0s before scheduled jobs daemon start
I230314 06:17:56.582368 2203155 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 678  tenant-settings-watcher: established range feed cache
I230314 06:17:56.588501 2204143 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 679  tenant-capability-watcher: established range feed cache
I230314 06:17:56.598863 2204550 13@kv/kvserver/replicate_queue.go:834  [T1,n1,replicate,s1,r58/1:/{Table/57-Max}] 680  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230314 06:17:56.602891 2206300 13@kv/kvserver/replicate_queue.go:834  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 681  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230314 06:17:56.603587 2204143 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:147  [T1,n1] 682  received results of a full table scan for tenant capabilities
I230314 06:17:56.613826 2194723 jobs/registry.go:1489  [T1,n1] 683  POLL JOBS STATS job 101: stepping through state running
I230314 06:17:56.625367 2202535 server/auto_upgrade.go:78  [T1,n1] 684  no need to upgrade, cluster already at the newest version
I230314 06:17:56.653398 2213020 13@kv/kvserver/replicate_queue.go:834  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 685  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230314 06:17:56.663806 2202508 sql/syntheticprivilegecache/cache.go:195  [T1,n1] 686  warmed privileges for virtual tables in 84.697575ms
I230314 06:17:56.752369 2222971 13@kv/kvserver/replicate_queue.go:834  [T1,n1,replicate,s1,r4/1:/System{/tsd-tse}] 687  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230314 06:17:56.762701 1823186 1@server/server_sql.go:1627  [T1,n1] 688  serving sql connections
I230314 06:17:56.764567 1823186 testutils/testcluster/testcluster.go:1338  [T1] 689  WaitForFullReplication
I230314 06:17:56.765171 1823186 testutils/testcluster/testcluster.go:1342  [T1] 690  WaitForFullReplication took: 366ns
I230314 06:17:56.797686 2225277 util/log/event_log.go:32  [T1,n1,client=127.0.0.1:51010,user=root] 691 ={"Timestamp":1678774676784755611,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"kv.rangefeed.enabled\" = true","Tag":"SET CLUSTER SETTING","User":"root","SettingName":"kv.rangefeed.enabled","Value":"true"}
I230314 06:17:56.821452 2225277 util/log/event_log.go:32  [T1,n1,client=127.0.0.1:51010,user=root] 692 ={"Timestamp":1678774676803272298,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"kv.closed_timestamp.target_duration\" = '100ms'","Tag":"SET CLUSTER SETTING","User":"root","SettingName":"kv.closed_timestamp.target_duration","Value":"00:00:00.1"}
I230314 06:17:56.930124 2225277 5@util/log/event_log.go:32  [T1,n1,client=127.0.0.1:51010,user=root] 693 ={"Timestamp":1678774676830241862,"EventType":"create_table","Statement":"CREATE TABLE defaultdb.public.t ()","Tag":"CREATE TABLE","User":"root","DescriptorID":104,"TableName":"defaultdb.public.t"}
I230314 06:17:56.932721 1823186 spanconfig/spanconfigsqlwatcher/sqlwatcher.go:245  [T1] 694  established range feed over system.descriptors starting at time 1678774676.824485738,0
I230314 06:17:56.933386 1823186 spanconfig/spanconfigsqlwatcher/sqlwatcher.go:304  [T1] 695  established range feed over system.zones starting at time 1678774676.824485738,0
I230314 06:17:56.933674 1823186 spanconfig/spanconfigsqlwatcher/sqlwatcher.go:400  [T1] 696  established range feed over system.protected_ts_records starting at time 1678774676.824485738,0
    sqlwatcher_test.go:460: handler should never run
I230314 06:17:56.936109 1823186 testutils/testcluster/testcluster.go:148  [T1] 697  TestCluster quiescing nodes
I230314 06:17:56.936980 1828039 server/start_listen.go:103  [T1,n1] 698  server shutting down: instructing cmux to stop accepting
I230314 06:17:56.937128 1844941 sql/stats/automatic_stats.go:507  [T1,n1] 699  quiescing auto stats refresher
I230314 06:17:56.938219 1921194 rpc/context.go:2249  [T1,n1,rnode=1,raddr=127.0.0.1:41747,class=default,rpc] 700  connection heartbeat loop ended with err: <nil>
W230314 06:17:56.938498 2231051 kv/kvserver/intentresolver/intent_resolver.go:826  [T1] 701  failed to gc transaction record: could not GC completed transaction anchored at /Table/6/1/"kv.closed_timestamp.target_duration"/0: node unavailable; try another peer
I230314 06:17:56.943069 2148143 jobs/registry.go:1489  [T1,n1] 702  KEY VISUALIZER job 100: stepping through state succeeded
E230314 06:17:56.944306 2235869 kv/kvclient/kvcoord/txn_interceptor_committer.go:489  [T1] 703  making txn commit explicit failed for "sql txn" meta={id=4b684a05 key=/Table/3/1/104/2/1 pri=0.01860429 epo=0 ts=1678774676.830241862,0 min=1678774676.830241862,0 seq=3} lock=true stat=STAGING rts=1678774676.830241862,0 wto=false gul=1678774677.330241862,0 ifw=2: result is ambiguous: server shutdown
E230314 06:17:56.946724 1845349 jobs/registry.go:965  [T1,n1] 704  error expiring job sessions: expire-sessions: failed to read query result: query execution canceled
E230314 06:17:56.947134 1845349 jobs/registry.go:973  [T1,n1] 705  failed to serve pause and cancel requests: txn exec: context canceled
W230314 06:17:56.947305 1845349 jobs/registry.go:1049  [T1,n1] 706  canceling all adopted jobs due to stopper quiescing
W230314 06:17:56.950381 2235945 1@sql/event_log.go:653  [T1,n1,client=127.0.0.1:51010,user=root] 707  unable to save 1 entries to system.eventlog: log-event: failed to read query result: query execution canceled
W230314 06:17:56.939153 1840017 sql/sqlliveness/slinstance/slinstance.go:312  [T1,n1] 708  exiting heartbeat loop
W230314 06:17:56.959703 1840017 sql/sqlliveness/slinstance/slinstance.go:297  [T1,n1] 709  exiting heartbeat loop
E230314 06:17:56.959763 1840017 server/server_sql.go:477  [T1,n1] 710  failed to run update of instance with new session ID: node unavailable; try another peer
W230314 06:17:56.965222 2148143 jobs/adopt.go:461  [T1,n1] 711  could not clear job claim: clear-job-claim: node unavailable; try another peer
W230314 06:17:56.975362 2194723 jobs/adopt.go:461  [T1,n1] 712  could not clear job claim: clear-job-claim: failed to send RPC: sending to all replicas failed; last error: ba: Scan [/Table/15/1/101,/Table/15/1/102), [txn: b31fd413], [can-forward-ts] RPC error: node unavailable; try another peer
--- FAIL: TestSQLWatcherOnEventError (5.01s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #97319 pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test: TestSQLWatcherOnEventError failed [C-test-failure O-robot T-kv branch-release-22.2]
- #82031 pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test: TestSQLWatcherOnEventError failed [C-test-failure O-robot branch-release-22.1 sync-me sync-me-5]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test.TestSQLWatcherOnEventError [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9071198?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9071198?buildTab=artifacts#/) on master @ [9b43dc50d468171bc9919dc684b7274b63e13da2](https://github.com/cockroachdb/cockroach/commits/9b43dc50d468171bc9919dc684b7274b63e13da2):


```
I230315 06:20:13.225633 2067962 jobs/adopt.go:247  [T1,n1] 662  job 100: resuming execution
I230315 06:20:13.239162 1745769 jobs/wait.go:142  [T1,n1] 663  waited for 1 [848005735232503809] queued jobs to complete 360.040468ms
I230315 06:20:13.240498 1745769 upgrade/upgrademanager/manager.go:268  [T1,n1] 664  running permanent upgrade for version 1000022.2-38
I230315 06:20:13.342239 2070264 jobs/registry.go:1489  [T1,n1] 665  KEY VISUALIZER job 100: stepping through state running
I230315 06:20:13.347661 1745769 upgrade/upgrademanager/manager.go:585  [T1,n1] 666  running Upgrade to 1000022.2-38: "create jobs metrics polling job"
I230315 06:20:13.348066 2079450 jobs/adopt.go:247  [T1,n1] 667  job 848005736826699777: resuming execution
I230315 06:20:13.359573 2081067 jobs/adopt.go:247  [T1,n1] 668  job 848005736826699777: resuming execution
I230315 06:20:13.432421 2083608 jobs/registry.go:1489  [T1,n1] 669  MIGRATION job 848005736826699777: stepping through state running
I230315 06:20:13.545161 2083608 jobs/registry.go:1489  [T1,n1] 670  MIGRATION job 848005736826699777: stepping through state succeeded
I230315 06:20:13.546257 1771613 jobs/adopt.go:110  [T1,n1] 671  claimed 1 jobs
I230315 06:20:13.637376 1745769 jobs/wait.go:142  [T1,n1] 672  waited for 1 [848005736826699777] queued jobs to complete 289.328757ms
I230315 06:20:13.637495 1745769 server/server_sql.go:1542  [T1,n1] 673  done ensuring all necessary startup migrations have run
I230315 06:20:13.645654 2110249 jobs/job_scheduler.go:407  [T1,n1] 674  waiting 4m0s before scheduled jobs daemon start
I230315 06:20:13.652257 2111111 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 675  tenant-settings-watcher: established range feed cache
I230315 06:20:13.668737 2112335 13@kv/kvserver/replicate_queue.go:834  [T1,n1,replicate,s1,r58/1:/{Table/57-Max}] 677  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230315 06:20:13.668191 2113083 jobs/adopt.go:247  [T1,n1] 676  job 101: resuming execution
I230315 06:20:13.670211 2113445 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 678  tenant-capability-watcher: established range feed cache
I230315 06:20:13.695952 2110398 server/auto_upgrade.go:78  [T1,n1] 679  no need to upgrade, cluster already at the newest version
I230315 06:20:13.696447 2116180 13@kv/kvserver/replicate_queue.go:834  [T1,n1,replicate,s1,r12/1:/Table/{8-11}] 680  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230315 06:20:13.697356 2113445 multitenant/tenantcapabilities/tenantcapabilitieswatcher/watcher.go:147  [T1,n1] 681  received results of a full table scan for tenant capabilities
I230315 06:20:13.698739 2117313 13@kv/kvserver/replicate_queue.go:834  [T1,n1,replicate,s1,r52/1:/Table/5{1-2}] 682  error processing replica: 0 of 1 live stores are able to take a new replica for the range (1 already has a voter, 0 already have a non-voter); likely not enough nodes in cluster
I230315 06:20:13.705070 2110387 sql/syntheticprivilegecache/cache.go:195  [T1,n1] 683  warmed privileges for virtual tables in 57.923954ms
I230315 06:20:13.792983 2118743 jobs/registry.go:1489  [T1,n1] 684  POLL JOBS STATS job 101: stepping through state running
I230315 06:20:13.805597 1745769 1@server/server_sql.go:1629  [T1,n1] 685  serving sql connections
I230315 06:20:13.829329 1745769 testutils/testcluster/testcluster.go:1338  [-] 686  WaitForFullReplication
I230315 06:20:13.829417 1745769 testutils/testcluster/testcluster.go:1342  [-] 687  WaitForFullReplication took: 265ns
I230315 06:20:13.869103 2134159 util/log/event_log.go:32  [T1,n1,client=127.0.0.1:44982,user=root] 688 ={"Timestamp":1678861213851667615,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"kv.rangefeed.enabled\" = true","Tag":"SET CLUSTER SETTING","User":"root","SettingName":"kv.rangefeed.enabled","Value":"true"}
I230315 06:20:13.901197 2134159 util/log/event_log.go:32  [T1,n1,client=127.0.0.1:44982,user=root] 689 ={"Timestamp":1678861213871292216,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"kv.closed_timestamp.target_duration\" = '100ms'","Tag":"SET CLUSTER SETTING","User":"root","SettingName":"kv.closed_timestamp.target_duration","Value":"00:00:00.1"}
I230315 06:20:14.019665 2134159 5@util/log/event_log.go:32  [T1,n1,client=127.0.0.1:44982,user=root] 690 ={"Timestamp":1678861213902285655,"EventType":"create_table","Statement":"CREATE TABLE defaultdb.public.t ()","Tag":"CREATE TABLE","User":"root","DescriptorID":104,"TableName":"defaultdb.public.t"}
I230315 06:20:14.020865 1745769 spanconfig/spanconfigsqlwatcher/sqlwatcher.go:245  [-] 691  established range feed over system.descriptors starting at time 1678861213.902025550,0
I230315 06:20:14.021013 1745769 spanconfig/spanconfigsqlwatcher/sqlwatcher.go:304  [-] 692  established range feed over system.zones starting at time 1678861213.902025550,0
I230315 06:20:14.021149 1745769 spanconfig/spanconfigsqlwatcher/sqlwatcher.go:400  [-] 693  established range feed over system.protected_ts_records starting at time 1678861213.902025550,0
    sqlwatcher_test.go:460: handler should never run
I230315 06:20:14.025135 1745769 testutils/testcluster/testcluster.go:148  [-] 694  TestCluster quiescing nodes
I230315 06:20:14.026516 1751303 server/start_listen.go:103  [T1,n1] 695  server shutting down: instructing cmux to stop accepting
I230315 06:20:14.026967 1771547 sql/stats/automatic_stats.go:507  [T1,n1] 696  quiescing auto stats refresher
W230315 06:20:14.027638 2137520 kv/kvserver/intentresolver/intent_resolver.go:826  [-] 697  failed to gc transaction record: could not GC completed transaction anchored at /Table/15/1/101/0: node unavailable; try another peer
E230315 06:20:14.027846 2146280 kv/kvclient/kvcoord/txn_interceptor_committer.go:489  [-] 698  making txn commit explicit failed for "sql txn" meta={id=e5c4cbc8 key=/Table/3/1/104/2/1 pri=0.02715676 epo=0 ts=1678861213.902285655,0 min=1678861213.902285655,0 seq=3} lock=true stat=STAGING rts=1678861213.902285655,0 wto=false gul=1678861214.402285655,0 ifw=2: result is ambiguous: server shutdown
I230315 06:20:14.029564 1853916 rpc/context.go:2249  [T1,n1,rnode=1,raddr=127.0.0.1:45891,class=default,rpc] 699  connection heartbeat loop ended with err: <nil>
W230315 06:20:14.029983 1765108 sql/sqlliveness/slinstance/slinstance.go:312  [n1] 701  exiting heartbeat loop
W230315 06:20:14.030041 1765108 sql/sqlliveness/slinstance/slinstance.go:297  [n1] 702  exiting heartbeat loop
E230315 06:20:14.030609 1765108 server/server_sql.go:477  [n1] 703  failed to run update of instance with new session ID: node unavailable; try another peer
I230315 06:20:14.028799 2070264 jobs/registry.go:1489  [T1,n1] 700  KEY VISUALIZER job 100: stepping through state succeeded
E230315 06:20:14.033367 1771613 jobs/registry.go:993  [T1,n1] 704  error claiming jobs: could not query jobs table: claim-jobs: failed to read query result: query execution canceled
E230315 06:20:14.035046 1771603 jobs/registry.go:965  [T1,n1] 705  error expiring job sessions: expire-sessions: failed to read query result: query execution canceled
E230315 06:20:14.035460 1771603 jobs/registry.go:973  [T1,n1] 706  failed to serve pause and cancel requests: txn exec: context canceled
W230315 06:20:14.035818 2146350 1@sql/event_log.go:653  [T1,n1,client=127.0.0.1:44982,user=root] 707  unable to save 1 entries to system.eventlog: log-event: failed to read query result: query execution canceled
W230315 06:20:14.036383 1771603 jobs/registry.go:1049  [T1,n1] 708  canceling all adopted jobs due to stopper quiescing
E230315 06:20:14.036719 2118743 jobs/registry.go:921  [T1,n1] 709  error getting live session: node unavailable; try another peer
--- FAIL: TestSQLWatcherOnEventError (4.48s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #97319 pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test: TestSQLWatcherOnEventError failed [C-test-failure O-robot T-kv branch-release-22.2]
- #82031 pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/spanconfigsqlwatcherccl_test: TestSQLWatcherOnEventError failed [C-test-failure O-robot branch-release-22.1 sync-me sync-me-5]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLWatcherOnEventError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/sqlwatcher_test.go
+++ b/pkg/ccl/spanconfigccl/spanconfigsqlwatcherccl/sqlwatcher_test.go
@@ -437,18 +437,18 @@ func TestSQLWatcherOnEventError(t *testing.T) {
 	tdb.Exec(t, `SET CLUSTER SETTING kv.rangefeed.enabled = true`)
 	tdb.Exec(t, `SET CLUSTER SETTING kv.closed_timestamp.target_duration = '100ms'`)
 
-	noopCheckpointDuration := time.Hour // effectively disable no-op checkpoints
 	sqlWatcher := spanconfigsqlwatcher.New(
 		keys.SystemSQLCodec,
 		ts.ClusterSettings(),
 		ts.RangeFeedFactory().(*rangefeed.Factory),
 		1<<20, /* 1 MB, bufferMemLimit */
 		ts.Stopper(),
-		noopCheckpointDuration,
+		time.Second, // doesn't matter
 		&spanconfig.TestingKnobs{
 			SQLWatcherOnEventInterceptor: func() error {
 				return errors.New("boom")
 			},
+			SQLWatcherSkipNoopCheckpoints: true,
 		},
 	)
 

--- a/pkg/spanconfig/spanconfigsqlwatcher/sqlwatcher.go
+++ b/pkg/spanconfig/spanconfigsqlwatcher/sqlwatcher.go
@@ -170,7 +170,8 @@ func (s *SQLWatcher) watch(
 			if err != nil {
 				return err
 			}
-			if len(sqlUpdates) == 0 && !checkpointNoops.ShouldProcess(timeutil.Now()) {
+			if len(sqlUpdates) == 0 &&
+				(!checkpointNoops.ShouldProcess(timeutil.Now()) || s.knobs.SQLWatcherSkipNoopCheckpoints) {
 				continue
 			}
 			if err := handler(ctx, sqlUpdates, combinedFrontierTS); err != nil {

--- a/pkg/spanconfig/testing_knobs.go
+++ b/pkg/spanconfig/testing_knobs.go
@@ -88,6 +88,10 @@ type TestingKnobs struct {
 	// often the SQLWatcher checkpoints noops.
 	SQLWatcherCheckpointNoopsEveryDurationOverride time.Duration
 
+	// SQLWatcherSkipNoopCheckpoints allows tests to skip no-op checkpoints
+	// entirely.
+	SQLWatcherSkipNoopCheckpoints bool
+
 	// SplitterStepLogger is used to capture internal steps the splitter is
 	// making, for debugging and test-readability purposes.
 	SplitterStepLogger func(string)

